components/admin_console/save_button.jsx

```diff
@@ -11,26 +11,28 @@ export default class SaveButton extends React.Component {
     static get propTypes() {
         return {
             saving: PropTypes.bool.isRequired,
-            disabled: PropTypes.bool
+            disabled: PropTypes.bool,
+            savingMessageId: PropTypes.string
         };
     }
 
     static get defaultProps() {
         return {
-            disabled: false
+            disabled: false,
+            savingMessageId: 'admin.saving'
         };
     }
 
     render() {
-        const {saving, disabled, ...props} = this.props; // eslint-disable-line no-use-before-define
+        const {saving, disabled, savingMessageId, ...props} = this.props; // eslint-disable-line no-use-before-define
 
         let contents;
         if (saving) {
             contents = (
                 <span>
                     <span className='icon fa fa-refresh icon--rotate'/>
                     <FormattedMessage
-                        id='admin.saving'
+                        id={savingMessageId}
                         defaultMessage='Saving Config...'
```

Comment:
Can you also pass in the default message as well? It's what we do elsewhere, and it'll make this work better in languages where translation isn't 100% completed

Reply:
Sorry I don't really understand, the defaultMessage attribute is there so what should I modify here exactly ?

Reply:
I mean we should pass the defaultMessage as a prop into the SaveButton so that if we're using it in different places, the defaultMessage will always match the id that's passed in. I don't know if we do it much in the web app, but we have a few places in the mobile app where we have something like this
```javascript
export default class MyComponents extends React.Component {
    static propTypes = {
        textId: PropTypes.string,
        textDefaultMessage: PropTypes.string
    };

    render() {
        return (
            <FormattedMessage
                id={this.props.id}
                defaultMessage={this.props.defaultMessage}
            />
         );
    }
}
```

Reply:
ok I see! Thanks for the clarification, I am gonna change it :)